Update documentation and structure (#136)
micro-ROS/rmw-microxrcedds: README.md

This package provides a middleware implementation for XRCE-DDS (rmw layer).
The implementation wraps the latest code from eProsima's Micro XRCE-DDS client to communicate with the DDS world.
This library defines the interface used by upper layers in the ROS 2 stack, and that is implemented using XRCE-DDS middleware in the lower layers.

#### Library configuration

This RMW implementation can be configured via CMake arguments, its usual to configure them via `colcon.meta` file in a micro-ROS enviroment.

> Most of these configuration are related to memory management because this RMW implementation tries to fully rely on static memory assignations. Because of this, assignations of the memory are upper bounded so must be configured by the user before the build process.

Reviewer: ```suggestion
Most of these configuration are related to memory management because this RMW implementation tries to fully rely on static memory assignations. This leads to an upper bound in memory assignations, which is configured by the user before the build process.
```

By default, the package sets the values for all memory bounded. The upper bound is configurable by a file that sets the values during the build process.

More details about RMW Micro XRCE-DDS can be found [here](https://micro.ros.org/docs/tutorials/advanced/microxrcedds_rmw_configuration/).

> s

Reviewer: ```suggestion
```

All the configurable parameters are:

| Name                                      | Description                                                                                                                                                                                    | Default |
| ----------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------- |
| RMW_UXRCE_TRANSPORT                       | Sets Micro XRCE-DDS transport to use. (udp, serial, custom)                                                                                                                                    | udp     |
| RMW_UXRCE_IPV                             | Sets Micro XRCE-DDS IP version to use. (ipv4, ipv6)                                                                                                                                            | ipv4    |

> | RMW_UXRCE_CREATION_MODE                   | Sets creation mode in Micro XRCE-DDS. (xml, refs)                                                                                                                                              | xml     |

Reviewer: is binary mode supported by the rmw?

Author: c0d6373a91ab0cb76ef5eefd0ed972cc62a77812
